biome.json

```diff
@@ -125,18 +125,13 @@
   "json": {
     "formatter": {
       "enabled": true
+    },
+    "parser": {
+      "allowComments": true,
+      "allowTrailingCommas": true
     }
```

Comment:
Looks like this is allowing json files to be created that will not be parsable

Re-reading the docs the hunk below wasn't needed, see: https://biomejs.dev/guides/how-biome-works/#json-like-files, but this seems to be not helpful for folks either. The linter should complain about comments and auto-fix trailing commas before tests start to fail.

Here's a PR and the CI run to demonstrate: 
- https://github.com/getsentry/sentry/pull/72014
- https://github.com/getsentry/sentry/actions/runs/9370576639/job/25797888214?pr=72014

Would also need to be fixed in https://github.com/getsentry/getsentry/pull/14185

Reply:
@ryan953 I personally think this should be handled by the build process, but I'm ok with reverting this change as well. Just FYI, any additions to `overrides` impact the execution time of Biome.

Reply:
How would or could the build process handle this? I'm not able to imagine what you mean

Reply:
Certainly a rules change like this probably shouldn't be hidden inside a PR that purports to update a dependency. So if the purpose was to remove the overrides and improve perf that's probably something to be done independently and to measure. I would assume that the perf difference will amount to a fraction of a percent of CI time

Reply:
> How would or could the build process handle this? 

Whoever loads the content of the JSON file and tries to parse it should and will throw an error if they don't support it. We can keep an explicit list of what is supported and what is not, but most libraries do support JSON files with comment. 

> Certainly a rules change like this probably shouldn't be hidden inside a PR that purports to update a dependency. So if the purpose was to remove the overrides and improve perf that's probably something to be done independently and to measure. I would assume that the perf difference will amount to a fraction of a percent of CI time

Thanks for the feedback 👍